apps/src/lab2/progress/README.md

```diff
@@ -0,0 +1,23 @@
+## Progress
+
+This contains a new progress system that can be used by any **Lab2** lab.  It allows for a level to define a variety of conditions, some of which give the user feedback, and some of which allow the user to continue to the next level.
+
+It was first introduced by Music Lab in https://github.com/code-dot-org/code-dot-org/pull/50596.  Adapted from that PR's description, here are some key features:
+
+- Each level has a set of validation conditions.
+- The level designer writes these in JSON, rather than JavaScript, so we have a clean separation between per-level validation conditions and the lab's internal implementation.
+- Conditions that aren't recognised are skipped, allowing level designers to add necessary conditions while lab engineers will later add support for them.
+- Level designers can specify specific feedback messages for specific conditions.
```

Comment:
very nitty: using 'specific/specify' 3x in this sentence

Reply:
Hah, good point.